ddtrace/tracer/remote_config.go

```diff
@@ -425,3 +498,48 @@ func (t *tracer) startRemoteConfig(rcConfig remoteconfig.ClientConfig) error {
 
 	return nil
 }
+
+func (t *tracer) startDynamicInstrumentationRCSubscriptions() error {
+	t.dynInstMu.Lock()
+	defer t.dynInstMu.Unlock()
```

Comment:
I get that code above this layer should only call this if the subscription is currently off (and similarly the stop call should only be called if there already was a subscription (because you're checking whether an update occurred), but should that be either documented or should this code be made more defensive by checking if there are non-zero tokens already?